quarto/_extensions/countdown/countdown.lua

```diff
@@ -0,0 +1,353 @@
+  local styleTable = {}
+  
+  -- Build the style
+  for i, key in ipairs(styleKeys) do
+    styleTable[i] = safeStyle(options, key)
+  end
+
+  -- Concatenate entries together
+  return table.concat(styleTable)
+end
+
+local function structureCountdownCSSVars(options)
+  local dots = {}
+
+  for key, value in pairs(options) do
+      table.insert(dots, string.format("--countdown-%s: %s;", key:gsub("_", "-"), value))
+  end
+
+  return dots
+end
+
+local function countdown_style(options, defaults)
+
+  -- Check if options have values; if it is empty, just exit.
+  if isVariableEmpty(options) or isTableEmpty(options) then
+    return nil
+  end
+
+  -- Determine the selector value
+  local possibleSelector = getOption(options, "selector", defaults.selector)
+
+  -- Begin CSS Variables
+  local cssTable = {}
+
+  -- Pass defaults into make_countdown_css
+  for key, defaultValue in pairs(defaults) do
+    -- Assign into the CSS table if key is present
+    cssTable[key] = getOption(options, key, defaultValue)
+  end
+
+  -- Delete the selector key (after the fact)
+  cssTable["selector"] = nil
+
+  -- Restructure options to ("--countdown-<key>: <value>;)
+  local structuredCSS = structureCountdownCSSVars(cssTable)
+
+  -- Embed into the document to avoid rendering to disk and, then, embedding a URL.
+  quarto.doc.include_text('in-header', 
+    string.format(
+      "<!-- Countdown Global CSS -->\n<style text='text/css'>:%s {%s}</style>", 
+      possibleSelector,
+      table.concat(structuredCSS)
+    )
+  )
+  -- Note: This feature or using `add_supporting` requires Quarto v1.4 or above
+
+end
+
+-- Handle embedding/creation of assets once
+local function ensureHTMLDependency(meta)
+
+  -- Register _all_ assets together.
+  quarto.doc.addHtmlDependency({
+    name = "countdown",
+    version = countdownEmbeddedVersion,
+    scripts = { "assets/countdown.js"},
+    stylesheets = { "assets/countdown.css"},
+    resources = {"assets/smb_stage_clear.mp3"}
+  })
+
+  -- Embed custom settings into the document based on document-level settings
+  countdown_style(meta.countdown, default_style)
+
+  -- Disable re-exporting if no-longer needed
+  needsToExportDependencies = false
+end
+
+-- Process unnamed time string format
+local function parseTimeString(args)
+  if #args == 0 or type(args[1]) ~= "string" then
+    return nil
+  end
+
+  local minutes, seconds = args[1]:match("(%d+):(%d+)")
+  if minutes == nil or seconds == nil then
+    quarto.log.error(
+      "The quartodown time string must be in the format 'MM:SS'.\n" ..
+      "Please correct countdown timer with time string given as `" .. args[1] .. "`"
+    )
+    assert("true" == "false")
+  end
+
+  return { minutes = tonumber(minutes), seconds = tonumber(seconds) }
+end
+
+local function countdown(args, kwargs, meta)
+  local minutes, seconds
+
+  -- Retrieve named time arguments and fallback on default values if missing
+  arg_time = parseTimeString(args)
```

Comment:
I think I missed a local here

```suggestion
  local arg_time = parseTimeString(args)
```

Reply:
One of the downsides to Lua; everything is global without using `local` 😞